src/server/replica.h

```diff
@@ -165,6 +171,9 @@ class Replica {
   facade::RespVec resp_args_;
   facade::CmdArgVec cmd_str_args_;
 
+  std::optional<Context> cntx_;     // Context for tasks in replica.
+  ::boost::fibers::mutex cntx_mu_;  // guards updating cntx_ with new instances.
```

Comment:
The reason for having a context at all and not letting Stop tear everything down forcefully is:
- we need a context either way to propagate errors from flow fibers
- we might need to do partial stops in the future, where we keep the error contained, fail one stage (like the full sync) and then retry. Letting the resource owner do its cleanup is easier is a better approach in this case